### PR TITLE
[API] Bounds-check that memory for instance is above a minimum, meets a granularity (#1030)

### PR DESCRIPTION
**Pull request by ryaeng**
- Require the user to select a memory size of at least one gibibyte
    when creating a new instance.
- Create new test to match this requirement.
- Update tests that created instances under one gibibyte.

### CODE REVIEW

**Comment by smklein** [comment]
Hey @ryaeng , thanks for this PR!

Couple pieces of feedback:
- As mentioned in https://github.com/oxidecomputer/omicron/pull/1150 , could we make this a constant value, such that we can tweak the exact boundary later, if necessary?
- I know it's not documented externally as well as it could be, but I think we were leaning towards a [256 MiB](https://github.com/oxidecomputer/omicron/blob/90acf64a892eed821f267eb5a540c7aaaff9e19c/common/src/api/external/mod.rs#L329-L330) granularity for memory, rather than 1 GiB. Hopefully by using a constant value, this becomes easy to tweak.

File: nexus/src/app/instance.rs [hunk 1]
```diff
@@ -56,6 +56,16 @@ impl super::Nexus {
             )));
         }
 
+        // Reject instances where the memory is not at least one gibibyte
+        if params.memory.to_whole_gibibytes() < 1 {
```
Comment (smklein) [inline nexus/src/app/instance.rs:60]: To close out https://github.com/oxidecomputer/omicron/issues/1030 , could we do a granularity check too?

**Review by smklein — APPROVED** [review]
Looks good! Merging now

**Comment by ryaeng** [comment]
> Looks good! Merging now

Thanks, Sean!

**Pull request merged**